docs/deploy_streamlit_app.md

```diff
@@ -1,3 +1,156 @@
+```
+
+- If you have requirements for apt-get, add them to `packages.txt`, one package name per line. See our streamlit-apps demo repo for an [example packages.txt file](https://github.com/streamlit-apps/ml-projects/blob/master/packages.txt).
+
+## Log in to share.streamlit.io
+
+The first thing you'll see is a button to login with GitHub. Please login with the primary email associated with your GitHub account.
+
+```eval_rst
+.. important:: If the email you originally signed-up with isn't the primary email associated with your GitHub account, just reply to your invite email telling us your primary Github email so we can grant access to the correct account.
+```
+
+You can find your [GitHub account email](https://github.com/settings/emails) here:
+<img src="_static/img/GitHub_email.png" alt="Github email" width="450" style='display: block; margin-left: auto; margin-right: auto; padding-top: 15px;'/>
+
+## Deploy your app
+
+Click "New app", then fill in your repo, branch, and file path, and click "Deploy". Your app will take a minute or two to deploy and then you'll be ready to share!
+
+If your app has a lot of dependencies it may take a few minutes to deploy the first time. But after that, any change that does not touch your dependencies should show up immediately!
+
+That's it — you're done! Once your app is deployed it will be available at
+
+```python
+https://share.streamlit.io/[user name]/[repo name]/[branch name]/[app path]
+```
+
+for example:
+
+```python
+http://share.streamlit.io/streamlit/demo-self-driving/master/streamlit_app.py
+```
+
+Also for your URL, the branch and app path are not required, if there's only one matching app in that repo. So for the above example, you could also use the shortened URL: [http://share.streamlit.io/streamlit/demo-self-driving](http://share.streamlit.io/streamlit/demo-self-driving)
+
+## Share, update, and collaborate
+
+Now that your app is deployed you can easily share it and collaborate on it. But first, let's take a moment and do a little joy dance for getting that app deployed! 🕺💃
+
+### Share your app
+
+Your app is now live at that fixed URL, so go wild and share it with whomever you want. From your deployed app you can click on the "☰" menu on the top right and select 'Share this app' to post it directly into social media or to share with the community on the [Forum](https://discuss.streamlit.io/c/streamlit-examples/9). We'd love to see what you make!
+
+### Update your app
+
+Your GitHub repository is the source for the app, so that means that any time you push an update to your repo you'll see it reflected in the app in almost real time. Try it out!
+
+Streamlit also smartly detects whether you touched your dependencies, in which case it will automatically do a full redeploy for you - which will take a little more time. But since most updates don't involve dependency changes, you should usually see your app update in real time.
```

Comment:
Not sure if we want to call this out here but we've disabled the auto rerun on save so now users will have to trigger a rerun to see the latest changes.